New History page

ProgDaniel opened the pull request:
The History page has been created. On it, you can view prior recommended playlists and expand them to see the tracks that have been recommended.

Tests for relevant pages created

(Not yet available: Turn a recommended playlist into a Spotify Playlist by tapping the Spotify logo)

ProgDaniel committed:
History Page

ProgDaniel committed:
WIP: History Page

Pop-up now appears when tapping a playlist; content not yet available on it.

ProgDaniel committed:
History Page - View Playlist

Playlists in the History Page can now be opened to see its tracks

Tests for History Page, History Playlist Page created

ProgDaniel committed:
Renamed 'Recommendations' to 'History'

The sidedrawer now reads 'History' instead of 'Recommendations'

ProgDaniel committed:
History - Removed Splash Effect on Playlists

Removed splash effect that occurs when holding on a playlist in the History Page

Formalized a non-implemented warning message

ProgDaniel commented:
![](https://github.trello.services/images/mini-trello-icon.png) [history page](https://trello.com/c/zfko9VTr/98-history-page)

LivHolm commented on lib/utilities/general_util.dart:
```diff
@@ -7,6 +7,20 @@ import 'package:smartphone_app/values/values.dart';
+        " " +
+        hour.toString() +
+        ":" +
+        minute.toString();
```
![image](https://user-images.githubusercontent.com/43985985/164445310-5f4e4a12-cfd1-4937-9da6-c825799e17a5.png)

Vi skal have fundet en måde at sætte et 0 foran minut tallet, når der kun er et tal.

ProgDaniel replied:
Dette er blevet rettet i det nyeste commit

LivHolm commented on lib/pages/history/history_page_ui.dart:
```diff
@@ -0,0 +1,139 @@
+          onTap: () {
+            bloc.add(PlaylistSelected(selectedPlaylist: playlist));
+          },
+          leading: const Icon(Icons.broken_image,
```
Jeg tænker de billeder der bliver vist i location dropdown menuen, kan blive brugt her, alt efter hvilken locations type det handler om.

ProgDaniel replied:
Done i den nyeste version

LivHolm reviewed (commented):
Alle test virker, og UI virker til at være fint, bortset fra mine kommentarer. Skal funktionaliteten ikke skal med i denne branch?

ProgDaniel committed:
Reformatted DateTime on History

Format is now DD-MM-YYYY HH:MM, and will always display the appropriate number of digits

ProgDaniel committed:
Merge remote-tracking branch 'origin/fix_play_button_functionality' into history_page

ProgDaniel committed:
Merge branch 'history_page' into history_page_alt

ProgDaniel committed:
Location Icons added to History Page

The location type of each playlist is now visible in the icon.

Restructured and redesigned both the History Page & History Playlist Page

Modified Mock Playlists for varied results

ProgDaniel committed:
History Page Documentation

ProgDaniel committed:
Massive refactoring and systematic changes

changed 'S' to 's'

ProgDaniel committed:
Merge branch 'develop' into history_page

ProgDaniel committed:
Added 'songs' to localizations

ProgDaniel committed:
Merge branch 'history_page' of https://github.com/cs-22-sw-8-07/smartphone_app into history_page

ProgDaniel committed:
Add null check to setString in app_values_helper
